wasm/sdk.go

```diff
@@ -0,0 +1,800 @@
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			var preferredBlobbers []string
+
+			err := sdk.InitStorageSDK(clientJSON,
+				Configuration.BlockWorker,
+				Configuration.ChainID,
+				Configuration.SignatureScheme,
+				preferredBlobbers)
+			// TODO: there is a problem when testing `zcncore.Init()` function
+			// especially in the `zcncore.GetNetworkDetails()`
+			// err := initSDK(clientJSON, chainJSON)
+			if err != nil {
+				reject.Invoke(err.Error())
+			}
+
+			resolve.Invoke(true)
+		}()
+
+		// SdkInitialized need to set to 'true' if err is nil.
+		// This will be available when other function that need sdk
+		// to be preset is called in WASM
+		sdk.SdkInitialized = true
+		return nil
+	})
+
+	promiseConstructor := js.Global().Get("Promise")
+	return promiseConstructor.New(handler)
+}
+
+func initSDK(clientJSON string, chainJSON string) error {
+	if len(Configuration.BlockWorker) == 0 ||
+		len(Configuration.ChainID) == 0 ||
+		len(Configuration.SignatureScheme) == 0 {
+		return NewError("invalid_param", "Configuration is empty")
+	}
+
+	var preferredBlobbers []string
+
+	err := sdk.InitStorageSDK(clientJSON,
+		Configuration.BlockWorker,
+		Configuration.ChainID,
+		Configuration.SignatureScheme,
+		preferredBlobbers)
+
+	if err != nil {
+		return err
+	}
+
+	return zcncore.Init(chainJSON)
+}
+
+func InitAuthTicket(this js.Value, p []js.Value) interface{} {
+	authTicket := p[0].String()
+	result := sdk.InitAuthTicket(authTicket)
+	return result
+}
+
+func SetSDKLogLevel(this js.Value, p []js.Value) interface{} {
+	logLevel, _ := strconv.Atoi(p[0].String())
+
+	sdk.SetLogLevel(logLevel)
+	return nil
+}
+
+func SetSDKLogFile(this js.Value, p []js.Value) interface{} {
+	logFile := p[0].String()
+	verbose, _ := strconv.ParseBool(p[1].String())
+
+	sdk.SetLogFile(logFile, verbose)
+	return nil
+}
+
+func GetNetwork(this js.Value, p []js.Value) interface{} {
+	n := sdk.GetNetwork()
+	result, err := json.Marshal(n)
+
+	if err != nil {
+		return map[string]interface{}{
+			"error": fmt.Sprintf("GetNetwork failed. Reason: %s", err),
+		}
+	}
+
+	return string(result)
+}
+
+func SetMaxTxnQuery(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetMaxTxnQuery(num)
+	return nil
+}
+
+func SetQuerySleepTime(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetQuerySleepTime(num)
+	return nil
+}
+
+func SetMinSubmit(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetMinSubmit(num)
+	return nil
+}
+
+func SetMinConfirmation(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetMinConfirmation(num)
+	return nil
+}
+
+func ZBOXSetNetwork(this js.Value, p []js.Value) interface{} {
+	miners := []string{p[0].String()}
+	sharders := []string{p[1].String()}
+	sdk.SetNetwork(miners, sharders)
+	return nil
+}
+
+// //
```

Comment:
@Rizary Pls cleanup all comments, they should be only on Top of the function explaining what this function is for.